browser/ts/index.ts

```diff
@@ -0,0 +1,370 @@
+    const zeroExMesh: ZeroExMesh;
+}
+
+/**
+ * A set of configuration options for Mesh.
+ */
+export interface Config {
+    // The URL of an Ethereum node which supports the Ethereum JSON RPC API.
+    // Used to validate and watch orders.
+    ethereumRPCURL: string;
+    // The network ID to use when communicating with Ethereum.
+    ethereumNetworkID: number;
+    // UseBootstrapList is whether to use the predetermined list of peers to
+    // bootstrap the DHT and peer discovery. Defaults to true.
+    useBootstrapList?: boolean;
+    // The amount of time (in seconds) before the order's stipulated expiration
+    // time that it will be considered expired. Higher values will cause orders
+    // to be considered invalid sooner. Defaluts to 10.
+    orderExpirationBufferSeconds?: number;
+    // The polling interval (in seconds) to wait before checking for a new
+    // Ethereum block that might contain transactions that impact the
+    // fillability of orders stored by Mesh. Different networks have different
+    // block producing intervals: POW networks are typically slower (e.g.,
+    // Mainnet) and POA networks faster (e.g., Kovan) so one should adjust the
+    // polling interval accordingly. Defaults to 5.
+    blockPollingIntervalSeconds?: number;
+    // The maximum request Content-Length accepted by the backing Ethereum RPC
+    // endpoint used by Mesh. Geth & Infura both limit a request's content
+    // length to 1024 * 512 Bytes. Parity and Alchemy have much higher limits.
+    // When batch validating 0x orders, we will fit as many orders into a
+    // request without crossing the max content length. The default value is
+    // appropriate for operators using Geth or Infura. If using Alchemy or
+    // Parity, feel free to double the default max in order to reduce the number
+    // of RPC calls made by Mesh. Defaults to 524288 bytes.
+    ethereumRPCMaxContentLength?: number;
+}
+
+// The global entrypoint for creating a new MeshWrapper.
+interface ZeroExMesh {
+    newWrapperAsync(config: Config): Promise<MeshWrapper>;
+}
+
+// A direct translation of the MeshWrapper type in Go. Its API exposes only
+// simple JavaScript types like number and string, some of which will be
+// converted. For example, we will convert some strings to BigNumbers.
+interface MeshWrapper {
+    startAsync(): Promise<void>;
+    onError(handler: (err: Error) => void): void;
+    onOrderEvents(handler: (events: Array<MeshOrderEvent>) => void): void;
+    addOrdersAsync(orders: Array<MeshSignedOrder>): Promise<MeshValidationResults>;
+}
+
+// The type for signed orders exposed by MeshWrapper. Unlike other types, the
+// analog isn't defined here. Instead we re-use the definition in
+// @0x/order-utils.
+interface MeshSignedOrder {
+    makerAddress: string;
+    makerAssetData: string;
+    makerAssetAmount: string;
+    makerFee: string;
+    takerAddress: string;
+    takerAssetData: string;
+    takerAssetAmount: string;
```

Comment:
Why aren't amounts parsed into BigNumbers?

Reply:
I think I made this pretty clear in the comments for `MeshWrapper` and `MeshSignedOrder`:

```ts
// A direct translation of the MeshWrapper type in Go. Its API exposes only
// simple JavaScript types like number and string, some of which will be
// converted. For example, we will convert some strings to BigNumbers.
interface MeshWrapper
```

```ts
// The type for signed orders exposed by MeshWrapper. Unlike other types, the
// analog isn't defined here. Instead we re-use the definition in
// @0x/order-utils.
interface MeshSignedOrder
```

`MeshSignedOrder` describes the type of something that comes directly from Go/Wasm. We can't (easily) convert strings to `BigNumber` in Go/Wasm so instead we do it here. The types that are actually exposed by the `@0x/mesh-browser` package are `BigNumber`s where appropriate. Take a look at the `meshSignedOrderToSignedOrder` function for an example of where the conversion happens.

Do you feel there is something missing from the comments or is there anything I can do to help make this more clear?

Reply:
The comment above `MeshWrapper` is clear to me. Given the "it's API exposes only simple Javascript types..." part, it wasn't immediately obvious to me that this included `MeshSignedOrder`. But now it's clear. I think we can probably just leave it, I must have skimmed through it too quickly.